<a href="https://colab.research.google.com/github/bonitr02/datasci_3_eda/blob/main/HHA507_datasci_loading.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Loading Packages

In [137]:
import pandas as pd
import re
import numpy as np
import time
import polars as pl

# <b> <u> HHA 507 Assignment 3: Exploratory Data Analysis in Health Informatics

In [ ]:
### Loading in packages


### Load in data


### Look for missingness


### Look for outliers


### Univariate descriptives: (descriptives, frequencies, histograms, etc.)


### Bivariate descriptives: (correlations, covariances)

# <b> <u> HHA 507 Assignment 2: Data Manipulation in Python: Pandas, Polars, Dask, and Modin


## Data Cleaning with Pandas

### Loading and Previewing'Healthcare Data Cleaning' Dataset

Original Dataset

In [63]:
healthcare = pd.read_csv('/content/healthcare_data_cleaning (1).csv')


Prior to data cleaning:

Number of rows: 105,000

Number of columns: 20

In [83]:
# Check the size and shape of the DataFrame. After data cleaning, this value will change.
print("After data cleaning, Number of rows:", healthcare.shape[0])
print("After data cleaning, Number of columns:", healthcare.shape[1])

After data cleaning, Number of rows: 35828
After data cleaning, Number of columns: 20


### Handling Missing Values

#### Identify values entered as "missing" in the dataset

In [84]:
# Check for missing values in a column. Run before and after removing "missing" data values.
healthcare['Gender'].value_counts()

Other     12050
Male      11905
Female    11873
Name: Gender, dtype: int64

In [85]:
# Check for missing values in a column. Run before and after removing "missing" data values.
healthcare['Payment Method'].value_counts()

Cash         11996
Card         11921
Insurance    11911
Name: Payment Method, dtype: int64

In [86]:
# Check for missing values in a column. Run before and after removing "missing" data values.
healthcare['Patient Age'].value_counts()

18    1573
42    1024
47     977
44     974
40     938
      ... 
94       4
97       3
98       3
91       3
99       2
Name: Patient Age, Length: 83, dtype: int64

In [88]:
# Check for missing values in a column. Run before and after removing "missing" data values.
healthcare['City of Residence'].value_counts()

West Michael         40
Lake Michael         35
East Michael         31
North Michael        30
West James           27
                     ..
Vaughanfort           1
Gilbertport           1
Wallaceton            1
North Kennethfort     1
Lake Lisachester      1
Name: City of Residence, Length: 20260, dtype: int64

In [89]:
# Check for missing values in a column. Run before and after removing "missing" data values.
healthcare['Disease Diagnosed'].value_counts()

Cold        7228
Flu         7185
None        7162
Allergy     7142
Covid-19    7111
Name: Disease Diagnosed, dtype: int64

#### Remove values entered as "missing" in the dataset

In [71]:
for x in healthcare.index:
  if healthcare.loc[x, "Payment Method"] == 'missing' or healthcare.loc[x, "Gender"] == 'missing' or healthcare.loc[x, "Patient Age"] == 'missing' or healthcare.loc[x,'Preferred Doctor'] == 'missing' or healthcare.loc[x,'Medication Prescribed'] == 'missing' or healthcare.loc[x,'Type of Appointment'] == 'missing' or healthcare.loc[x,'Average Heart Rate']  == 'missing' or healthcare.loc[x,'Average BP'] == 'missing' or healthcare.loc[x, 'Prescription Cost ($)'] == 'missing' :
    healthcare.drop(x, inplace = True)



In [70]:
for x in healthcare.index:
  if healthcare.loc[x, "City of Residence"] == 'missing' or healthcare.loc[x,'State of Residence']  == 'missing' or healthcare.loc[x,'Has Insurance']  == 'missing' or healthcare.loc[x,'Visited Last Month']  == 'missing' or healthcare.loc[x,'Disease Diagnosed']  == 'missing' or healthcare.loc[x,'Height (in cm)']  == 'missing' or healthcare.loc[x,'Weight (in kg)']  == 'missing' or healthcare.loc[x,'Payment Due ($)']  == 'missing' or healthcare.loc[x,'Last Visit (days ago)']  == 'missing' or healthcare.loc[x,'Visit Duration (mins)']  == 'missing' or healthcare.loc[x,'Number of Tests'] == 'missing':
    healthcare.drop(x, inplace = True)

#### Drop rows with any missing values

In [72]:
print(healthcare.isnull().sum())
# No missing values identified in this dataset

Patient Age              0
Gender                   0
City of Residence        0
State of Residence       0
Has Insurance            0
Visited Last Month       0
Payment Method           0
Preferred Doctor         0
Disease Diagnosed        0
Medication Prescribed    0
Type of Appointment      0
Average Heart Rate       0
Average BP               0
Height (in cm)           0
Weight (in kg)           0
Payment Due ($)          0
Last Visit (days ago)    0
Visit Duration (mins)    0
Number of Tests          0
Prescription Cost ($)    0
dtype: int64


### Remove duplicate rows and columns

In [73]:
healthcare.drop_duplicates(inplace=True)

### Clean column names
Remove white spaces and special characters from column names

In [74]:
healthcare.columns

Index(['Patient Age', 'Gender', 'City of Residence', 'State of Residence',
       'Has Insurance', 'Visited Last Month', 'Payment Method',
       'Preferred Doctor', 'Disease Diagnosed', 'Medication Prescribed',
       'Type of Appointment', 'Average Heart Rate', 'Average BP',
       'Height (in cm)', 'Weight (in kg)', 'Payment Due ($)',
       'Last Visit (days ago)', 'Visit Duration (mins)', 'Number of Tests',
       'Prescription Cost ($)'],
      dtype='object')

In [75]:
healthcare_cleaned = healthcare.rename(columns={"Patient Age":"PatientAge","State of Residence":"StateOfResidence","City of Residence":"CityOfResidence","Has Insurance":"HasInsurance","Visted Last Month":"VisitedLastMonth", "Payment Method":"PaymentMethod","Preferred Doctor":"PreferredDoctor","Disease Diagnosed":"DiseaseDiagnosed", "Medication Prescribed":"MedicationPrescribed","Type of Appointment":"TypeOfAppointment","Average Heart Rate":"AverageHeartRate","Average BP":"AverageBP","Height (in cm)":"HeightInCM", "Weight (in kg)":"WeightInKg","Payment Due ($)":"PaymentDueInDollars","Last Visit (days ago)":"LastVisitDaysAgo","Visit Duration (mins)":"VisitDurationMins","Number of Tests":"NumberOfTests","Prescription Cost ($)":"PrescriptionCostInDollars"})


In [76]:
healthcare_cleaned

PatientAge  Gender   CityOfResidence StateOfResidence HasInsurance  \
0              45   Other        West Brian         Kentucky          Yes   
4              48  Female         Birdmouth          Montana           No   
7              52    Male      Vanessaburgh      Connecticut          Yes   
8              20  Female  West Briannabury          Montana           No   
11             40  Female        Jonesmouth     Rhode Island           No   
...           ...     ...               ...              ...          ...   
104992         44  Female    Port Tracistad           Alaska          Yes   
104993         42  Female     West Annaport          Florida          Yes   
104994         53  Female     Michelleburgh     South Dakota           No   
104995         74   Other      West Anthony         Oklahoma          Yes   
104997         51  Female  Lake Lisachester         Nebraska          Yes   

       Visited Last Month PaymentMethod PreferredDoctor DiseaseDiagnosed  \
0                     Yes     Insurance    Dr. Williams              Flu   
4                      No     Insurance       Dr. Brown             None   
7                     Yes     Insurance    Dr. Williams              Flu   
8                     Yes          Card       Dr. Jones              Flu   
11                     No          Card       Dr. Jones         Covid-19   
...                   ...           ...             ...              ...   
104992                 No     Insurance       Dr. Brown             Cold   
104993                 No          Cash     Dr. Johnson         Covid-19   
104994                 No          Card    Dr. Williams              Flu   
104995                 No          Card    Dr. Williams         Covid-19   
104997                Yes     Insurance    Dr. Williams          Allergy   

       MedicationPrescribed TypeOfAppointment AverageHeartRate AverageBP  \
0                     Med_E           General               66       111   
4                     Med_C        Specialist               93        89   
7                     Med_E           General               99       129   
8                     Med_C        Specialist               67       108   
11                    Med_E        Specialist               65       106   
...                     ...               ...              ...       ...   
104992                Med_A         Follow-Up               76        93   
104993                Med_D        Specialist               80       129   
104994                Med_B         Emergency               68       109   
104995                Med_E        Specialist               65       112   
104997                Med_D         Follow-Up               89       118   

       HeightInCM WeightInKg PaymentDueInDollars LastVisitDaysAgo  \
0             183         70         65.45721578              193   
4             186         87         264.2147372              188   
7             170         47         387.0850435              179   
8             173         86         172.5483428              174   
11            166         82         299.1426761              193   
...           ...        ...                 ...              ...   
104992        178         66         462.5656709              189   
104993        171         78         131.3563258              185   
104994        172         75         488.0316041              167   
104995        169         61         309.1644461              183   
104997        179         71          320.830418              172   

       VisitDurationMins NumberOfTests PrescriptionCostInDollars  
0                     92             5               15.71751736  
4                    113             3               77.48311305  
7                    108             2               76.96592633  
8                     91             2               142.3171077  
11                    92             1               77.71536034  
...                  ... 

### <b> Summary of Changes Made to the Dataset:

Original dataset: 105,000 rows and 20 columns

After removing missing values: 37,594 rows and 20 columns

After removing duplicate rows: 35,828 rows and 20 columns

## Data Transformation

###Create new columns based on existing ones

#### Confirm Data Types

In [105]:
# Confirm data types
datatypes = healthcare_cleaned.dtypes
print(datatypes)

PatientAge                    object
Gender                        object
CityOfResidence               object
StateOfResidence              object
HasInsurance                  object
Visited Last Month            object
PaymentMethod                 object
PreferredDoctor               object
DiseaseDiagnosed              object
MedicationPrescribed          object
TypeOfAppointment             object
AverageHeartRate               int64
AverageBP                     object
HeightInCM                    object
WeightInKg                    object
PaymentDueInDollars           object
LastVisitDaysAgo              object
VisitDurationMins              int64
NumberOfTests                 object
PrescriptionCostInDollars    float64
HeartRateStatus               object
ViralDiagnosis                object
dtype: object


#### Add a new column to designate Normal or High Average Heart Rates

In [ ]:
# Convert AverageHeartRate column to int
healthcare_cleaned['AverageHeartRate'] = healthcare_cleaned['AverageHeartRate'].apply(int)

# Create threshold value
threshold_high = 100

# Create a new column "HeartRateStatus" based on thresholds
healthcare_cleaned['HeartRateStatus'] = healthcare_cleaned['AverageHeartRate'].apply(lambda x: 'High' if x >= threshold_high else 'Normal')

# Confirm new column is added
print(healthcare_cleaned)

#### Add a new column to designate if the visit diagnoses was a viral illness

In [ ]:
# Define criteria for classifying diseases as "Viral"
viral_diseases = ['Covid-19', 'Flu', "Cold"]

# Create a new column "ViralDiagnosis" based on the criteria
healthcare_cleaned['ViralDiagnosis'] = healthcare_cleaned['DiseaseDiagnosed'].apply(lambda x: 'Viral' if x in viral_diseases else 'Non-Viral')

# Confirm new column is added
print(healthcare_cleaned)

### Aggregate data using groupby and compute summary statistics

#### Group average heart rates by patient age

In [91]:
grouped_hr = healthcare_cleaned.groupby('PatientAge').agg({
    'AverageHeartRate':'mean'
})
print(grouped_hr)

            AverageHeartRate
PatientAge                  
100                74.666667
18                 79.456453
19                 79.044444
20                 79.931408
21                 79.449123
...                      ...
95                 77.000000
96                 85.400000
97                 87.666667
98                 85.000000
99                 79.000000

[83 rows x 1 columns]


#### Group average prescription cost by diagnosed disease

In [100]:
# Convert PrescriptionCostInDollars column to float
healthcare_cleaned['PrescriptionCostInDollars'] = healthcare_cleaned['PrescriptionCostInDollars'].apply(float)

grouped_cost = healthcare_cleaned.groupby('DiseaseDiagnosed').agg({
    'PrescriptionCostInDollars':'mean'
})
print(grouped_cost)

                  PrescriptionCostInDollars
DiseaseDiagnosed                           
Allergy                           74.341221
Cold                              74.738794
Covid-19                          73.801153
Flu                               74.757829
None                              75.124267


In [101]:
# Sum average prescription cost by Disease Diagnosed
grouped_costSum = healthcare_cleaned.groupby('DiseaseDiagnosed').agg({
    'PrescriptionCostInDollars':'sum'
})
print(grouped_costSum)


                  PrescriptionCostInDollars
DiseaseDiagnosed                           
Allergy                            530945.0
Cold                               540212.0
Covid-19                           524800.0
Flu                                537135.0
None                               538040.0


### Use pivot tables or cross-tabulations for multi-dimensional analysis

#### Pivot Tablet: Average Visit Duration in Minutes by Number of Tests per Visit



In [143]:
# Convert PrescriptionCostInDollars column to float
healthcare_cleaned['VisitDurationMins'] = healthcare_cleaned['VisitDurationMins'].apply(int)

healthcare_cleaned['VisitLength'] = pd.cut(healthcare_cleaned['VisitDurationMins'], bins=[0, 30, 60, 120, 240, np.inf], labels=['0-30', '30-60', '60-120', '120-240', 'over 240'])
pivot_data = healthcare_cleaned.pivot_table(columns='VisitLength', index='NumberOfTests', values='VisitDurationMins', aggfunc='mean')
print(pivot_data)



VisitLength         0-30      30-60      60-120
NumberOfTests                                  
0              21.352941  45.813559   90.252525
1              19.958478  45.845361   89.628502
10             30.000000        NaN  110.000000
2              20.172500  45.614407   90.833333
3              19.950173  45.627561   90.736650
4              20.243888  45.227346   90.542114
5              20.306998  45.535497   90.949482
6              19.742099  45.795620   90.737736
7              20.295139  45.518817   89.605166
8              20.415385  45.717949   88.497608
9              18.076923  42.857143   91.192308


#### Pivot Table: Average Prescription Cost by Patient Insurance Status

In [132]:
healthcare_cleaned['RXCostInDollars'] = pd.cut(healthcare_cleaned['PrescriptionCostInDollars'], bins=[0, 20, 40, 60, 80, 100, 200, np.inf], labels=['0-20', '20-40', '40-60', '60-80', '80-100', '100-200', 'over 200'])
pivot_data1 = healthcare_cleaned.pivot_table(columns='RXCostInDollars', index='HasInsurance', values='PrescriptionCostInDollars', aggfunc='mean')
print(pivot_data1)


RXCostInDollars      0-20       2-40      40-60      60-80     80-100  \
HasInsurance                                                            
No               10.56106  30.789582  50.562581  70.554286  90.445957   
Yes              10.40166  30.376093  50.222360  70.424736  90.759882   

RXCostInDollars     100-200  
HasInsurance                 
No               124.849380  
Yes              125.041122  


## Introduction to Polar and Speed Test

In [138]:
# Define the dataset path
dataset_path = '/content/healthcare_data_cleaning (1).csv'

In [139]:
# Using Pandas
start_time = time.time()
pandas_df = pd.read_csv(dataset_path)
pandas_time = time.time() - start_time

In [140]:
# Using Polar
start_time = time.time()
polar_df = pl.read_csv(dataset_path)
polar_time = time.time() - start_time

In [141]:
# Print results and execution times
print(f"Pandas Execution Time: {pandas_time:.4f} seconds")
print(f"Polar Execution Time: {polar_time:.4f} seconds")

Pandas Execution Time: 0.2044 seconds
Polar Execution Time: 0.0674 seconds


### <b> Using Import Time Module:

Pandas Execution Time: 0.2044 seconds

Polar Execution Time: 0.0674 seconds

Polar load times are much faster than Pandas

#  <u> <b> HHA 507 Assignment 1: Using Google Colab and Acquiring Datasets  

## CSV Datasets

### CSV Dataset 1: Location of COVID-19 Testing Locations in the United States

#### Dataset Features


*   Provider Name: Name of COVID-19 test location provider
*   Address1: Primary address
*   Address2: Suite number
*   City: City of provider
*   State: State of provider
*   Zip: Zip code of provider
*   Last Report Date: Most recent date a test was reported as given
*   Public Website: Website address for testing provider
*   Phone Number: Phone number of testing provider
*   Geopoint: Geographic coordinates of testing location

In [ ]:
covid19_test_to_treat = pd.read_csv('/content/COVID-19_Test_to_Treat (2).csv')

In [ ]:
print (covid19_test_to_treat.columns)


Index(['Provider Name', 'Address1', 'Address2', 'City', 'State', 'Zip',
       'Last Report Date', 'Public Website', 'Phone Number', 'Geopoint'],
      dtype='object')


In [ ]:
covid19_test_to_treat.head(10)

Provider Name                   Address1      Address2  \
0            WALGREENS STORE #11795         3551 CASSOPOLIS ST           NaN   
1                  SAFEWAY PHARMACY      8646 Richmond Highway           NaN   
2                  CVS STORE #00947                 28 Abby Rd  CVS Pharmacy   
3              CHURCH AVENUE CLINIC            2412 Church Ave           NaN   
4          AFC Urgent Care Main St.              4200 Main St.           NaN   
5  GoHealth Urgent Care - Bruce Twp             80650 Van Dyke           NaN   
6          KROGER PHARMACY 02400705           890 RICHMOND PLZ           NaN   
7                  CVS STORE #09723           11428 KENYON WAY  CVS Pharmacy   
8            WALGREENS STORE #06191  10529 LOVELAND MADEIRA RD           NaN   
9           NH GoHealth Thomasville       1122 Randolph Street     Suite 110   

               City State    Zip        Last Report Date  \
0           ELKHART    IN  46514  08/23/2023 12:00:00 AM   
1        ALEXANDRIA    VA  22309  07/15/2023 12:00:00 AM   
2           PALMYRA    VA  22963  08/18/2023 12:00:00 AM   
3          BROOKLYN    NY  11226  08/24/2023 12:00:00 AM   
4        BRIDGEPORT    CT   6605  08/04/2023 12:00:00 AM   
5         BRUCE TWP    MI  48065  08/31/2023 12:00:00 AM   
6          RICHMOND    KY  40475  08/31/2023 12:00:00 AM   
7  RANCHO CUCAMONGA    CA  91701  08/18/2023 12:00:00 AM   
8          LOVELAND    OH  45140  08/23/2023 12:00:00 AM   
9       THOMASVILLE    NC  27360  08/15/2023 12:00:00 AM   

                                      Public Website  Phone Number  \
0        https://www.walgreens.com/findcare/services           NaN   
1                                                NaN           NaN   
2                    https://www.cvs.com/coronavirus           NaN   
3                                                NaN  718-681-8700   
4                                                NaN           NaN   
5                                                NaN           NaN   
6  https://www.kroger.com/health/pharmacy/covid-1...           NaN   
7                    https://www.cvs.com/coronavirus           NaN   
8  https://www.walgreens.com/locator/walgreens-10...           NaN   
9                                                NaN           NaN   

                       Geopoint  
0    POINT (-85.97459 41.73845)  
1    POINT (-77.12114 38.72421)  
2     POINT (-78.312535 37.899)  
3   POINT (-73.955365 40.65027)  
4  POINT (-73.124955 41.215805)  
5    POINT (-83.03924 42.88694)  
6     POINT (-84.27457 37.7317)  
7  POINT (-117.555695 34.13551)  
8  POINT (-84.273325 39.261295)  
9    POINT (-80.07466 35.85845)

### CSV Dataset 2: Behavioral Risks Associated with Tobacco Smoking

#### Dataset Features


*   Year: Smoking survey year
*   LocationAbbr: Abbreviated survey location
*   LocationDesc: Location of survey
*   TopicType: Name of survey
*   TopicDesc: Description of smoking implement
*   MeasureDesc: Survey smoking measure
*   DataSource: Source of data
*   Response: Survey response to smoking frequency
*   Data_Value_Unit: Unit of data value
*   Data_Value_Type: Description of data value unit
*   Data_Value: Numerical data value
*   Data_Value_Footnote_Symbol: Symbol of footnote
*   Data_Value_Footnote: Footnote corresponding to survey
*   Data_Value_Std_Err: Standard deviation of sample population
*   Low_Confidence_Limit: Lower value of confidence interval
*   High_Confidence_Limit: Higher value of confidence interval
*   Sample_Size: Survey population sample size
*   Gender: Gender of survey population
*   Race: Race of survey population
*   Age: Age range of survey population
*   Education: Education level of survey population
*   GeoLocation: Geographic coordinates of survey
*   TopicTypeID: ID of topic type
*   TopicId: Topic Type
*   MeasureId: Measure ID
*   StratificationID1: Internal stratification ID
*   StratificationID2: Internal stratification ID
*   StratificationID3: Internal stratification ID
*   StratificationID4: Internal stratification ID
*   SubMeasureID: Internal submeasure ID
*   DisplayOrder: Survey display order


In [ ]:
tobacco_use = pd.read_csv('/content/BehavioralRisk_Tobacco.csv')

In [ ]:
print(tobacco_use.columns)

Index(['YEAR', 'LocationAbbr', 'LocationDesc', 'TopicType', 'TopicDesc',
       'MeasureDesc', 'DataSource', 'Response', 'Data_Value_Unit',
       'Data_Value_Type', 'Data_Value', 'Data_Value_Footnote_Symbol',
       'Data_Value_Footnote', 'Data_Value_Std_Err', 'Low_Confidence_Limit',
       'High_Confidence_Limit', 'Sample_Size', 'Gender', 'Race', 'Age',
       'Education', 'GeoLocation', 'TopicTypeId', 'TopicId', 'MeasureId',
       'StratificationID1', 'StratificationID2', 'StratificationID3',
       'StratificationID4', 'SubMeasureID', 'DisplayOrder'],
      dtype='object')


In [ ]:
tobacco_use.head(10)

YEAR LocationAbbr                     LocationDesc  \
0       2017           GU                             Guam   
1       2018           US  National Median (States and DC)   
2       2017           US  National Median (States and DC)   
3       2016           GU                             Guam   
4       2014           GU                             Guam   
5       2012           IN                          Indiana   
6  2018-2019           DE                         Delaware   
7       2011           MO                         Missouri   
8       2013           ME                            Maine   
9       2017           WA                       Washington   

                   TopicType                       TopicDesc  \
0  Tobacco Use – Survey Data          Cigarette Use (Adults)   
1  Tobacco Use – Survey Data          Cigarette Use (Adults)   
2  Tobacco Use – Survey Data          Cigarette Use (Adults)   
3  Tobacco Use – Survey Data  Smokeless Tobacco Use (Adults)   
4  Tobacco Use – Survey Data          Cigarette Use (Adults)   
5  Tobacco Use – Survey Data          Cigarette Use (Adults)   
6  Tobacco Use – Survey Data  Smokeless Tobacco Use (Adults)   
7  Tobacco Use – Survey Data  Smokeless Tobacco Use (Adults)   
8  Tobacco Use – Survey Data          Cigarette Use (Adults)   
9  Tobacco Use – Survey Data          Cigarette Use (Adults)   

                              MeasureDesc DataSource   Response  \
0                         Current Smoking      BRFSS        NaN   
1                          Smoking Status      BRFSS    Current   
2                          Smoking Status      BRFSS      Never   
3                             Current Use      BRFSS        NaN   
4                         Current Smoking      BRFSS        NaN   
5                         Current Smoking      BRFSS        NaN   
6  Current Use – (2 yrs – Race/Ethnicity)      BRFSS        NaN   
7                             Current Use      BRFSS        NaN   
8                       Smoking Frequency      BRFSS  Some Days   
9                          Smoking Status      BRFSS    Current   

  Data_Value_Unit Data_Value_Type  ...  \
0               %      Percentage  ...   
1               %      Percentage  ...   
2               %      Percentage  ...   
3               %      Percentage  ...   
4               %      Percentage  ...   
5               %      Percentage  ...   
6               %      Percentage  ...   
7               %      Percentage  ...   
8               %      Percentage  ...   
9               %      Percentage  ...   

                                GeoLocation TopicTypeId TopicId  MeasureId  \
0                   (13.444304, 144.793731)         BEH  100BEH     110CSA   
1                                       NaN         BEH  100BEH     165SSA   
2                                       NaN         BEH  100BEH     165SSA   
3                   (13.444304, 144.793731)         BEH  150BEH     177SCU   
4                   (13.444304, 144.793731)         BEH  100BEH     110CSA   
5  (39.766910452000445, -86.14996019399968)         BEH  100BEH     110CSA   
6  (39.008830667000495, -75.57774116799965)         BEH  150BEH     177SCR   
7  (38.635790776000476, -92.56630005299968)         BEH  150BEH     177SCU   
8  (45.254228894000505, -68.98503133599962)         BEH  100BEH     166SSP   
9  (47.52227862900048, -120.47001078999972)         BEH  100BEH     165SSA   

   StratificationID1  StratificationID2  StratificationID3 StratificationID4  \
0               2GEN               8AGE               6RAC              6EDU   
1               1GEN               8AGE               6RAC              6EDU   
2               1GEN               8AGE               6RAC              6EDU   
3               1GEN               8AGE               4RAC              6EDU   
4               1GEN               8AGE               5RAC              6EDU   
5               3GEN               8AGE               6RAC              6EDU   


### CSV Dataset 3: Impaired Driving Death Rates by Age and Gender in 2012 and 2014

#### Dataset Features


*   State: State of impaired driving deaths
*   Location: State and geolocation
*   All Ages, 2012: Rate of death at all ages in 2012
*   All Ages, 2014: Rate of death at all ages in 2014
*   Ages 0-20, 2012: Rate of death from 0 to 20 in 2012
*   Ages 0-20, 2014: Rate of death from 0 to 20 in 2014
*   Ages 21-34, 2012: Rate of death from 21 to 34 in 2012
*   Ages 21-34, 2014: Rate of death from 21 to 34 in 2014
*   Ages 35+, 2012: Rate of death for 35 years and older in 2012
*   Ages 35+, 2014: Rate of death for 35 years and older in 2014
*   Male, 2012: Rate of death at all ages in 2012
*   Male, 2014: Rate of death at all ages in 2014
*   Female, 2012: Rate of death at all ages in 2012
*   Female, 2014: Rate of death at all ages in 2014






In [ ]:
impaired_driving = pd.read_csv('/content/Impaired_Driving_Death_Rate__by_Age_and_Gender__2012___2014__All_States.csv')


In [ ]:
print(impaired_driving.columns)

Index(['State ', 'Location', 'All Ages, 2012', 'All Ages, 2014',
       'Ages 0-20, 2012', 'Ages 0-20, 2014', 'Ages 21-34, 2012',
       'Ages 21-34, 2014', 'Ages 35+, 2012', 'Ages 35+, 2014', 'Male, 2012',
       'Male, 2014', 'Female, 2012', 'Female, 2014'],
      dtype='object')


In [ ]:
impaired_driving.head(10)

State                                  Location  All Ages, 2012  \
0    Connecticut      Connecticut\n(41.562661, -72.64984)             2.4   
1       Kentucky        Kentucky\n(37.645973, -84.774968)             4.0   
2   South Dakota    South Dakota\n(44.35313, -100.373529)             5.7   
3       Oklahoma        Oklahoma\n(35.472031, -97.521068)             5.6   
4  West Virginia  West, Virginia\n(38.665511, -80.712638)             5.6   
5       Virginia        Virginia\n(37.542681, -78.457889)             2.5   
6           Iowa            Iowa\n(42.469401, -93.816487)             3.2   
7      Louisiana       Louisiana\n(31.312663, -92.445678)             5.2   
8      Tennessee       Tennessee\n(35.680941, -85.774487)             4.6   
9       Missouri        Missouri\n(38.635791, -92.566299)             4.8   

   All Ages, 2014  Ages 0-20, 2012  Ages 0-20, 2014  Ages 21-34, 2012  \
0             2.7              NaN              NaN               5.9   
1             3.9              1.8              2.0               7.2   
2             5.6              NaN              NaN               NaN   
3             4.0              NaN              NaN              10.5   
4             4.7              NaN              NaN              12.9   
5             2.5              1.2              1.2               4.5   
6             3.0              NaN              NaN               5.9   
7             5.4              2.8              1.8               8.6   
8             4.1              1.6              NaN               8.0   
9             3.4              1.9              NaN               9.5   

   Ages 21-34, 2014  Ages 35+, 2012  Ages 35+, 2014  Male, 2012  Male, 2014  \
0               6.1             2.1             2.5         3.7         4.6   
1               7.8             3.7             3.4         6.5         6.5   
2               NaN             NaN             6.2         7.4         8.7   
3               7.4             5.8             3.9         9.3         6.0   
4              10.0             4.4             4.0         9.3         7.1   
5               5.1             2.6             2.3         4.1         3.9   
6               4.6             2.9             3.4         5.1         4.8   
7               9.4             5.1             5.7         8.3         9.1   
8               8.2             4.9             4.1         7.5         6.8   
9               6.9             4.4             3.2         7.8         5.6   

   Female, 2012  Female, 2014  
0           1.1           NaN  
1           1.5           1.3  
2           NaN           NaN  
3           1.9           2.1  
4           NaN           NaN  
5           0.9           1.2  
6           NaN           NaN  
7           2.4           1.9  
8           1.8           1.5  
9           1.8           1.3

## XLSX Datasets

### XLSX Dataset 1: Percentage of Drivers and Front Seat Passengers Wearing Seatbelts in 2012 and 2014 in the United States

#### Dataset Features


*   State: Reported state
*   2012: Percentage of seatbelt wearers in 2012
*   2014: Percentage of seatbelt wearers in 2014
*   Location: Geolocation of state





In [ ]:
seatbelts = pd.read_excel('/content/PercenageOfDrivers.xlsx')

In [ ]:
print(seatbelts.columns)

Index(['State', 2012, 2014, 'Location'], dtype='object')


In [ ]:
seatbelts.head(10)

State  2012  2014                                Location
0    Connecticut    87    85    Connecticut\n(41.544654, -72.651713)
1       Missouri    79    79       Missouri\n(38.636168, -92.566292)
2   North Dakota    81    81  North Dakota\n(47.475879, -100.121011)
3    Mississippi    83    78    Mississippi\n(32.738663, -89.537312)
4     California    96    97    California\n(37.640735, -121.002435)
5  New Hampshire    69    70  New Hampshire\n(43.657736, -71.500736)
6        Montana    76    74       Montana\n(47.062617, -109.430168)
7       Arkansas    72    74       Arkansas\n(34.748745, -92.275105)
8        Wyoming    77    79        Wyoming\n(43.23558, -108.110518)
9         Nevada    91    94        Nevada\n(39.491833, -117.070159)

### XLSX Dataset 2: Pfizer COVID-19 Vaccine Distribution Allocations by Jurisdiction

#### Dataset Features


*   Jurisdiction: Location of allocation
*   Week of Allocations: Week allocation deployed
*   1st Dose Allocations: Number of first dose allocations
*   2nd Dose Allocations: Number of second dose allocations





In [ ]:
covid_vaccine_allocation = pd.read_excel('/content/COVID19VaccineDistributions.xlsx')


In [ ]:
print(covid_vaccine_allocation.columns)

Index(['Jurisdiction', 'Week of Allocations', '1st Dose Allocations',
       '2nd Dose Allocations'],
      dtype='object')


In [ ]:
covid_vaccine_allocation.head(10)

Jurisdiction Week of Allocations  1st Dose Allocations  \
0    Connecticut          2021-06-21                 54360   
1          Maine          2021-06-21                 21420   
2  Massachusetts          2021-06-21                104580   
3  New Hampshire          2021-06-21                 21420   
4   Rhode Island          2021-06-21                 17280   
5        Vermont          2021-06-21                 11070   
6     New Jersey          2021-06-21                132210   
7       New York          2021-06-21                167670   
8  New York City          2021-06-21                128070   
9    Puerto Rico          2021-06-21                 52290   

   2nd Dose Allocations  
0                 54360  
1                 21420  
2                104580  
3                 21420  
4                 17280  
5                 11070  
6                132210  
7                167670  
8                128070  
9                 52290

### XLSX Dataset 3: COVID-19 Vaccination Demographics in the United States

#### Dataset Features


*   Date: Reported date
*   Demographic category: Demographic group of vaccine recipients
*   Administered Dose1: Number of administrations
*   Administered Dose1 pct known: Percentage of known administrations
*   Administered Dose1 pct US: Percentage of administrations in the US
*   Administered Dose1 pct age group: Percentage of administrations in age group
*   Series Complete Yes: Number of reported completed vaccine series
*   Series Complete Pop pct agegroup: Percentage of completed series within the age group
*   Series Complete Pop Pct known: Percentage of known completed series
*   Series Complete Pop Pct US: Percentage of completed series in the US
*   Booster Doses Vax pct agegroup: Percentage of booster doses reported in the age group
*   Booster Dooses Pop Pct known: Percentage of known booster doses reported
*   Booster Doses Vax Pct US: Percentage of booster doses reported in the US
*   Booster Doses Pop Pct known Last 14 Days: Percentage of booster doses reported in the last 14 days
*   Booster Doses Yes: Number of booster doses reported
*   Booster Doses Yes Last 14 Days: Number of booster doses reported in the last 14 days
*   Second Booster Vax pct agegroup: Percentage of second boosters reported in the age group
*   Second Booster Pop Pct known: Percentage of known second boosters reported
*   Second Booster Pop Pct US: Percentage of second boosters reported in the US
*   Second Booster Pop Pct known Last 14 days: Percentage of second boosters reported in the last 14 days
*   Second Booster: Number of second boosters reported
*   Second Booster last 14 Days: Number of second boosters reported in the last 14 days
*   Bivalent Booster: Number of bivalent boosters reported
*   Bivalent Booster Pop Pct agegroup: Percentage of bivalent boosters reported within the age group
*   Bivalent Booster Pop Pct known: Percentage of known bivalent boosters reported





In [ ]:
covid_vaccine_demo = pd.read_excel('/content/COVID19VaccinationDemographics.xlsx')


In [ ]:
print(covid_vaccine_demo.columns)

Index(['Date', 'Demographic_category', 'Administered_Dose1',
       'Administered_Dose1_pct_known', 'Administered_Dose1_pct_US',
       'Series_Complete_Yes', 'Administered_Dose1_pct_agegroup',
       'Series_Complete_Pop_pct_agegroup', 'Series_Complete_Pop_Pct_known',
       'Series_Complete_Pop_Pct_US', 'Booster_Doses_Vax_pct_agegroup',
       'Booster_Doses_Pop_Pct_known', 'Booster_Doses_Vax_Pct_US',
       'Booster_Doses_Pop_Pct_known_Last14Days', 'Booster_Doses_Yes',
       'Booster_Doses_Yes_Last14Days', 'Second_Booster_Vax_pct_agegroup',
       'Second_Booster_Pop_Pct_known', 'Second_Booster_Pop_Pct_US',
       'Second_Booster_Pop_Pct_known_Last14Days', 'Second_Booster',
       'Second_Booster_Last14Days', 'Bivalent_Booster',
       'Bivalent_Booster_Pop_Pct_agegroup', 'Bivalent_Booster_Pop_Pct_known'],
      dtype='object')


In [ ]:
covid_vaccine_demo.head(10)

Date Demographic_category  Administered_Dose1  \
0 2023-05-10      Race_eth_NHAIAN             1911855   
1 2023-05-10          Age_unknown                9344   
2 2023-05-10     Race_eth_NHAsian            13983704   
3 2023-05-10  Race_eth_NHMult_Oth            12665103   
4 2023-05-10     Race_eth_NHBlack            21157654   
5 2023-05-10             Sex_Male           128162118   
6 2023-05-10     Race_eth_NHNHOPI              666432   
7 2023-05-10       Ages_25-39_yrs            57069471   
8 2023-05-10       Ages_40-49_yrs            36300944   
9 2023-05-10       Ages_65-74_yrs            34959727   

   Administered_Dose1_pct_known  Administered_Dose1_pct_US  \
0                           0.9                        0.7   
1                           0.0                        0.0   
2                           6.8                        5.2   
3                           6.2                        4.7   
4                          10.3                        7.8   
5                          47.9                       47.4   
6                           0.3                        0.2   
7                          21.1                       21.1   
8                          13.4                       13.4   
9                          12.9                       12.9   

   Series_Complete_Yes  Administered_Dose1_pct_agegroup  \
0              1588653                             78.5   
1                 2491                              NaN   
2             12609000                             73.6   
3             11389487                              NaN   
4             18545870                             51.3   
5            109144882                             78.4   
6               598992                             71.8   
7             47567709                             83.4   
8             31261428                             89.0   
9             30816762                             95.0   

   Series_Complete_Pop_pct_agegroup  Series_Complete_Pop_Pct_known  \
0                              65.2                            0.9   
1                               NaN                            0.0   
2                              66.4                            6.9   
3                               NaN                            6.2   
4                              45.0                           10.1   
5                              66.8                           47.6   
6                              64.5                            0.3   
7                              69.6                           20.6   
8                              76.6                           13.6   
9                              95.0                           13.4   

   Series_Complete_Pop_Pct_US  ...  Booster_Doses_Yes_Last14Days  \
0                         0.7  ...                           914   
1                         0.0  ...                             1   
2                         5.5  ...                          6400   
3                         4.9  ...                          4837   
4                         8.0  ...                          7656   
5                        47.3  ...                         35823   
6                         0.3  ...                           234   
7                        20.6  ...                          9317   
8                        13.6  ...                          5562   
9                        13.4  ...                         15109   

   Second_Booster_Vax_pct_agegroup  Second_Booster_Pop_Pct_known  \
0                             50.1                           0.6   
1                              0.0                           0.0   
2                             56.2                           6.1   
3                             54.9                           6.1   
4                             47.9                           8.2   
5                             52.2                          44.7   
6                             49.7                 

## JSON Datasets

### JSON Dataset 1: Diabetes Indicators

#### Dataset Features

*   Yearstart: First data collection year
*   Yearend: Last data collection Year
*   Locationabbr: Abbreviation of location
*   Locationdesc: Location name
*   Datasource: Source of data
*   Topic: Diabetes as topic
*   Question: Question related to diabetes asked
*   Datavalueunit: Unit of data value
*   Datavaluetype: Type of data value
*   Datavalue: Data provided
*   Datavaluealt: additional value provided
*   Lowconfidencelimit: Lower end of confidence interval
*   Highconfidencelimit: Higher end of confidence interval
*   Stratificationcategory1: Stratification category of study
*   Stratification1: Stratification value provided
*   Locationid: Internal ID signifying location
*   Topicid: Internal ID signifying topic
*   Questionid: Internal ID signifying question
*   Datavaluetypeid: Internal ID signifying data value type
*   Stratificationcategoryid1: Internal ID signifying stratification category
*   Stratificationid1: Internal ID signifying stratification ID
*   Datavaluefootnotesymbol: Symbol representing study footnote
*   Datavaluefootnote: Foot note

In [ ]:
diabetes_indicators = pd.read_json('/content/DiabetesIndicators.json')

In [ ]:
print(diabetes_indicators.columns)

Index(['yearstart', 'yearend', 'locationabbr', 'locationdesc', 'datasource',
       'topic', 'question', 'datavalueunit', 'datavaluetype', 'datavalue',
       'datavaluealt', 'lowconfidencelimit', 'highconfidencelimit',
       'stratificationcategory1', 'stratification1', 'locationid', 'topicid',
       'questionid', 'datavaluetypeid', 'stratificationcategoryid1',
       'stratificationid1', 'datavaluefootnotesymbol', 'datavaluefootnote'],
      dtype='object')


In [ ]:
diabetes_indicators.head(10)

yearstart  yearend locationabbr locationdesc datasource     topic  \
0       2011     2011           NV       Nevada      BRFSS  Diabetes   
1       2016     2016           NV       Nevada      BRFSS  Diabetes   
2       2019     2019           NV       Nevada      BRFSS  Diabetes   
3       2019     2019           NV       Nevada      BRFSS  Diabetes   
4       2017     2017           NE     Nebraska      BRFSS  Diabetes   
5       2015     2015           NE     Nebraska      BRFSS  Diabetes   
6       2018     2018           NE     Nebraska      BRFSS  Diabetes   
7       2011     2011           NV       Nevada      BRFSS  Diabetes   
8       2016     2016           NE     Nebraska      BRFSS  Diabetes   
9       2018     2018           NV       Nevada      BRFSS  Diabetes   

                                            question datavalueunit  \
0  Prevalence of diagnosed diabetes among adults ...             %   
1  Visits to dentist or dental clinic among adult...             %   
2  Prevalence of diagnosed diabetes among adults ...             %   
3  Foot examination among adults aged >= 18 years...             %   
4  Prevalence of high cholesterol among adults ag...             %   
5  Influenza vaccination among noninstitutionaliz...             %   
6  Influenza vaccination among noninstitutionaliz...             %   
7  Foot examination among adults aged >= 18 years...             %   
8  Pneumococcal vaccination among noninstitutiona...             %   
9  Dilated eye examination among adults aged >= 1...             %   

             datavaluetype  datavalue  ...  stratificationcategory1  \
0  Age-adjusted Prevalence       10.0  ...                  Overall   
1         Crude Prevalence       57.8  ...                   Gender   
2         Crude Prevalence       11.2  ...                   Gender   
3         Crude Prevalence        NaN  ...                   Gender   
4         Crude Prevalence       56.9  ...                  Overall   
5         Crude Prevalence       71.7  ...                   Gender   
6  Age-adjusted Prevalence       43.0  ...                   Gender   
7         Crude Prevalence        NaN  ...           Race/Ethnicity   
8  Age-adjusted Prevalence       37.8  ...                  Overall   
9  Age-adjusted Prevalence        NaN  ...                   Gender   

       stratification1  locationid topicid questionid  datavaluetypeid  \
0              Overall          32     DIA     DIA2_1       AGEADJPREV   
1                 Male          32     DIA     DIA8_0          CRDPREV   
2                 Male          32     DIA     DIA2_1          CRDPREV   
3                 Male          32     DIA     DIA5_0          CRDPREV   
4              Overall          31     DIA    DIA11_1          CRDPREV   
5               Female          31     DIA    DIA12_2          CRDPREV   
6               Female          31     DIA    DIA12_1       AGEADJPREV   
7  Other, non-Hispanic          32     DIA     DIA5_0          CRDPREV   
8              Overall          31     DIA    DIA13_1       AGEADJPREV   
9                 Male          32     DIA     DIA7_0       AGEADJPREV   

  stratificationcategoryid1 stratificationid1 datavaluefootnotesymbol  \
0                   OVERALL               OVR                     NaN   
1                    GENDER              GENM                     NaN   
2                    GENDER              GENM                     NaN   
3                    GENDER              GENM                       -   
4                   OVERALL               OVR                     NaN   
5                    GENDER              GENF                     NaN   
6                    GENDER              GENF                     NaN   
7                      RACE               OTH                       -   
8                   OVERALL               OVR                     NaN   
9                    GENDER              GENM                       -   

   datavaluefootnote  
0              

### JSON Dataset 2: Smoking Attributable Expenditures

#### Dataset Features

*   Year: Year of data
*   Locationabbr: Abbreviation of location
*   Locationdesc: Name of location
*   Datasource: Source of data
*   Topictype: Name of Topic
*   Topicdesc: Description of topic
*   Measuredesc: Type of measures described
*   Variable: Type of variable
*   Data_value_unit: Type of data value unit
*   Data_value_type: Description of data value
*   Data_value: Numerical data value
*   Data_value_footnote_symbol: Symbol to indicate footnote
*   Data_value_footnote: Footnote in study
*   Geolocation: Geographic location coordinates
*   Topic_type_id: Internal ID indicating topic type
*   Topic_id: Internal ID indicating topic
*   Measure_id: Internal ID indicating measure
*   Submeasureid: Internal ID indicating submeasure
*   Displayorder: Internal ID indicating display order

In [ ]:
smoking_expenditures = pd.read_json('/content/SAMEC.json')


In [ ]:
print(smoking_expenditures.columns)

Index(['year', 'locationabbr', 'locationdesc', 'datasource', 'topictype',
       'topicdesc', 'measuredesc', 'variable', 'data_value_unit',
       'data_value_type', 'data_value', 'data_value_footnote_symbol',
       'data_value_footnote', 'geolocation', 'topic_type_id', 'topic_id',
       'measure_id', 'submeasureid', 'displayorder'],
      dtype='object')


In [ ]:
smoking_expenditures.head(10)

year locationabbr   locationdesc datasource                      topictype  \
0  2009           RI   Rhode Island     SAMMEC  Health Consequences and Costs   
1  2007           SD   South Dakota     SAMMEC  Health Consequences and Costs   
2  2009           NH  New Hampshire     SAMMEC  Health Consequences and Costs   
3  2006           MD       Maryland     SAMMEC  Health Consequences and Costs   
4  2009           LA      Louisiana     SAMMEC  Health Consequences and Costs   
5  2006           TN      Tennessee     SAMMEC  Health Consequences and Costs   
6  2005           WI      Wisconsin     SAMMEC  Health Consequences and Costs   
7  2005           LA      Louisiana     SAMMEC  Health Consequences and Costs   
8  2007           MO       Missouri     SAMMEC  Health Consequences and Costs   
9  2008           TX          Texas     SAMMEC  Health Consequences and Costs   

                                  topicdesc      measuredesc  \
0  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
1  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
2  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
3  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
4  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
5  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
6  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
7  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
8  Smoking-Attributable Expenditures (SAEs)  Type of Expense   
9  Smoking-Attributable Expenditures (SAEs)  Type of Expense   

             variable data_value_unit      data_value_type  data_value  \
0               Other               $  Millions of Dollars        47.2   
1        Nursing Home               $  Millions of Dollars        29.2   
2               Other               $  Millions of Dollars        41.4   
3  Prescription Drugs               $  Millions of Dollars       465.2   
4               Other               $  Millions of Dollars       117.5   
5  Prescription Drugs               $  Millions of Dollars       555.8   
6          Ambulatory               $  Millions of Dollars       376.0   
7  Prescription Drugs               $  Millions of Dollars       319.0   
8  Prescription Drugs               $  Millions of Dollars       503.5   
9            Hospital               $  Millions of Dollars      4369.6   

  data_value_footnote_symbol  \
0                          *   
1                        NaN   
2                          *   
3                        NaN   
4                          *   
5                        NaN   
6                        NaN   
7                        NaN   
8                        NaN   
9                        NaN   

                                 data_value_footnote  \
0  Other expenditures include home health care, d...   
1                                                NaN   
2  Other expenditures include home health care, d...   
3                                                NaN   
4  Other expenditures include home health care, d...   
5                                                NaN   
6                                                NaN   
7                                                NaN   
8                                                NaN   
9                                                NaN   

                                         geolocation topic_type_id topic_id  \
0  {'latitude': '41.708280193', 'longitude': '-71...           HLT   501HLT   
1  {'latitude': '44.353130053', 'longitude': '-10...           HLT   501HLT   
2  {'latitude': '43.655950113', 'longitude': '-71...           HLT   501HLT   
3  {'latitude': '39.290580964', 'longitude': '-76...           HLT   501HLT   
4  {'latitude': '31.312660644', 'longitude': '-92...           HLT   501HLT   
5  {'latitude': '35.68094058', 'longitude': '-85....           HLT   501HLT   
6  {'latitude': '44.393191174', 'longitude': '-89...           HLT  

### JSON Dataset 3: Nutrition and Obesity Legislation by State

#### Dataset Features

*   Year: Year data published
*   Quarter: Quarter of the year data published
*   Locationabbr: Abbreviation of location name
*   Locationdesc: Location Name
*   Topicdesc: Health topic category
*   Policytopic: Policy topic
*   Datasource: Source of data
*   Setting: Type of setting data describes
*   Title: Title of study
*   Status: Status of policy implementation
*   Citation: Citation code
*   Statusaltvalue: Alternate status value
*   Datatype: Type of data
*   Comments: Abstract summary
*   Geolocation: Geographic coordinates of policy location
*   Displayorder: Order of display
*   Policytypeid: ID of policy type
*   Healthtopicid: ID of health topic
*   Policytopicid: ID of policy topic
*   Settingid: ID of setting
*   Provisionid: ID of provision
*   Enacteddate: Date policy enacted
*   Effectivedate: Effective date of policy


In [ ]:
obesity_legislation = pd.read_json('/content/ObesityLegislation.json')


In [ ]:
print(obesity_legislation.columns)

Index(['year', 'quarter', 'locationabbr', 'locationdesc', 'topicdesc',
       'policytopic', 'datasource', 'setting', 'title', 'status', 'citation',
       'statusaltvalue', 'datatype', 'comments', 'geolocation', 'displayorder',
       'policytypeid', 'healthtopicid', 'policytopicid', 'settingid',
       'provisionid', 'enacteddate', 'effectivedate'],
      dtype='object')


In [ ]:
obesity_legislation.head(10)

year  quarter locationabbr    locationdesc          topicdesc  \
0  2010        1           NY        New York            Obesity   
1  2011        1           WV   West Virginia          Nutrition   
2  2011        1           MA   Massachusetts  Physical Activity   
3  2010        1           NC  North Carolina            Obesity   
4  2011        1           FL         Florida  Physical Activity   
5  2011        1           FL         Florida            Obesity   
6  2008        1           NH   New Hampshire            Obesity   
7  2009        1           NY        New York          Nutrition   
8  2010        1           HI          Hawaii          Nutrition   
9  2009        1           NY        New York          Nutrition   

                                 policytopic datasource              setting  \
0                                 Incentives      DNPAO            Community   
1                             Appropriations      DNPAO            Community   
2  Built Environment and Street-Scale Design      DNPAO            Community   
3                              Menu Labeling      DNPAO            Community   
4               Parks, Recreation and Trails      DNPAO  School/After School   
5                  Health Insurance Coverage      DNPAO            Community   
6                          Food Restrictions      DNPAO            Community   
7                             Appropriations      DNPAO            Community   
8                             Appropriations      DNPAO            Community   
9                        Nutrition Standards      DNPAO            Community   

                           title status  ...  \
0              State Fiscal Plan   Dead  ...   
1              Sales Tax on Food   Dead  ...   
2              Pedestrian Safety   Dead  ...   
3               Governors Budget   Dead  ...   
4           Chronic Diseases<br>   Dead  ...   
5           Chronic Diseases<br>   Dead  ...   
6                  Poultry Sales   Dead  ...   
7  Nutrition and Fitness Website   Dead  ...   
8   Aliomanu Road Reconstruction   Dead  ...   
9  Nutrition and Fitness Website   Dead  ...   

                                            comments  \
0  (Abstract - Enacts into law major components o...   
1  (Abstract - Eliminates the sales tax on food e...   
2  (Abstract - Relates to pedestrian safety.   ),...   
3  (Abstract - Concerns governors budget. ), (Bil...   
4  (Abstract - "Creates the chronic disease preve...   
5  (Abstract - "Creates the chronic disease preve...   
6  (Abstract - Relates to the sale of poultry at ...   
7  (Abstract - Creates a nutrition and fitness we...   
8  (Abstract - Authorizes the issuance of General...   
9  (Abstract - Creates a nutrition and fitness we...   

                                         geolocation displayorder  \
0  {'latitude': '42.827001032', 'longitude': '-75...           38   
1  {'latitude': '38.665510202', 'longitude': '-80...            2   
2  {'latitude': '42.27687047', 'longitude': '-72....           68   
3  {'latitude': '35.466220975', 'longitude': '-79...           41   
4  {'latitude': '28.932040377', 'longitude': '-81...           59   
5  {'latitude': '28.932040377', 'longitude': '-81...           37   
6  {'latitude': '43.655950113', 'longitude': '-71...           35   
7  {'latitude': '42.827001032', 'longitude': '-75...            2   
8  {'latitude': '21.304850435', 'longitude': '-15...            2   
9  {'latitude': '42.827001032', 'longitude': '-75...            6   

  policytypeid healthtopicid  policytopicid settingid provisionid  \
0          LEG         002OB         029INC    002COM        1641   
1          LEG        001NUT         002APP    002COM        4781   
2          LEG         003PA         049BED    002COM        4090   
3          LEG         002OB         032MNL    002COM        2908   
4          LEG         003PA         034PRT    004SAS        5615   
5          LEG         002OB         028HIC    002COM